src/lib/isMACAddress.js

```diff
@@ -1,18 +1,34 @@
 import assertString from './util/assertString';
 
-const macAddress = /^(?:[0-9a-fA-F]{2}([-:\s]))([0-9a-fA-F]{2}\1){4}([0-9a-fA-F]{2})$/;
-const macAddressNoSeparators = /^([0-9a-fA-F]){12}$/;
-const macAddressWithDots = /^([0-9a-fA-F]{4}\.){2}([0-9a-fA-F]{4})$/;
+const macAddress48 = /^(?:[0-9a-fA-F]{2}([-:\s]))([0-9a-fA-F]{2}\1){4}([0-9a-fA-F]{2})$/;
+const macAddress48NoSeparators = /^([0-9a-fA-F]){12}$/;
+const macAddress48WithDots = /^([0-9a-fA-F]{4}\.){2}([0-9a-fA-F]{4})$/;
+const macAddress64 = /^(?:[0-9a-fA-F]{2}([-:\s]))([0-9a-fA-F]{2}\1){6}([0-9a-fA-F]{2})$/;
+const macAddress64NoSeparators = /^([0-9a-fA-F]){16}$/;
+const macAddress64WithDots = /^([0-9a-fA-F]{4}\.){3}([0-9a-fA-F]{4})$/;
 
 export default function isMACAddress(str, options) {
   assertString(str);
+  if (options?.eui) {
+    String(options.eui);
+  }
```

Comment:
```suggestion
  if (options?.eui) {
    options.eui = String(options.eui);
  }
```
I guess you meant to cast eui here